### ISSUE TITLE
Change poet client to cache the full response of the `v1/info`

### ISSUE THREAD

**Issue by fasmat** [spacemeshos/go-spacemesh#6226]
## Description
From a discussion on PR #6116: https://github.com/spacemeshos/go-spacemesh/pull/6116#discussion_r1706623480

The whole info response should be cached instead of just the `types.CertifierInfo` field of the response.

## Affected code
The poet client already caches part of the response returned from the `v1/info` endpoint via these two methods:

https://github.com/spacemeshos/go-spacemesh/blob/15ebc7b1881eaba3c6ac5f6c8a03126a2df10fc0/activation/poet.go#L618-L641

This should be changed so that the whole response is cached:

```go
func (c *poetService) getInfo(ctx context.Context) (*types.PoetInfo, error) {
	info, err := c.infoCache.get(func() (*types.PoetInfo, error) {
		info, err := c.client.Info(ctx)
		if err != nil {
			return nil, fmt.Errorf("getting info: %w", err)
		}
		return info, nil
	})
	if err != nil {
		return nil, err
	}
	return info, nil
}
```

simplifying `verifyPhaseShiftConfiguration` to the following:

```go
func (c *poetService) verifyPhaseShiftConfiguration(ctx context.Context) error {
	info, err := c.getInfo(ctx)
	if err != nil {
		return err
	}

	if info.PhaseShift != c.expectedPhaseShift {
		return ErrIncompatiblePhaseShift
	}

	return nil
}
```

**Comment by acud** [comment]
A note for consideration is [etags](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag). They're meant to allow for resource expiration at-the-server and allow intermediate layers to do caching and invalidation more easily. Just caching for some time might backfire under certain edge cases (please do consider my ignorance over the topic because I'm not sure they exist here).

**Comment by fasmat** [comment]
Sure, but this is just to keep track of a change that we excluded from a previous PR to get it merged quicker 🙂. PoET doesn't send Etags, the primary reason we are caching at all is to reduce the number of requests from the node to the servers. The node sends the same request for each identity to the same PoET (usually) around the same time, this just ensures that in that case we re-use the response we already got for the first identity for all others.

**Comment by poszu** [comment]
Implemented in #6274
